src/btreemap.rs

```diff
@@ -930,6 +930,62 @@ where
         }
     }
 
+    /// Returns an interator pointing to the first element below the specified key.
+    pub fn iter_upper_bound(&self, bound: &K) -> Iter<K, V, M> {
+        if self.root_addr == NULL {
+            // Map is empty.
+            return Iter::null(self);
+        }
+
+        let mut cursors = vec![];
+        let mut node = self.load_node(self.root_addr);
+        loop {
+            match node.keys.binary_search(bound) {
+                Ok(idx) | Err(idx) => {
+                    let child = match node.node_type {
```

Comment:
Do you really need to define `child`? We can probably just inline the code you have below in one match right away.

Reply:
Done!